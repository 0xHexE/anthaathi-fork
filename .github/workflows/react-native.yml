name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
