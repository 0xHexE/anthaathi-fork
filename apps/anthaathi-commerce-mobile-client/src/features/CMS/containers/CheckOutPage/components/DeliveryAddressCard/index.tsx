import React from 'react';
import {useIntl} from 'react-intl';
import {Pressable, View} from 'react-native';
import {Card, Text} from 'react-native-paper';
import {CheckOutPageComponentType} from '../../../../types/common';

export interface DeliveryAddressCardProps {
  deliveryTitle: string;
  deliveryAddress: string;
  buttonTitle: string;
  handlePress?: () => void;
  mobileNumber: string;
}

const DeliveryAddressCard = (props: DeliveryAddressCardProps) => {
  const intl = useIntl();
  return (
    <Card
      style={{
        marginHorizontal: 10,
        borderColor: '#E3E2E7',
        borderWidth: 1,
        borderRadius: 4,
        marginVertical: 5,
      }}
      testID="deliveryAddressCard">
      <Card.Content>
        <Text style={{color: '#364A15', fontSize: 16, fontWeight: '600'}}>
          {props.deliveryTitle}
        </Text>
        <View style={{flexDirection: 'row', marginVertical: 5}}>
          <Text style={{color: '#364A15', fontSize: 14, width: '85%'}}>
            {props.deliveryAddress}
          </Text>
          <Pressable
            testID="changeAddressButton"
            style={{paddingHorizontal: 10}}
            onPress={props.handlePress}>
            <Text
              testID="deliverAddressButtonTitle"
              style={{color: '#008D3E', fontSize: 14, fontWeight: '500'}}>
              {props.buttonTitle}
            </Text>
          </Pressable>
        </View>
        <View style={{flexDirection: 'row', marginVertical: 5}}>
          <Text style={{color: '#808080', fontSize: 14}}>
            {intl.formatMessage({defaultMessage: 'Mobile'})}
          </Text>
          <Text style={{color: '#364A15', fontSize: 14}}>
            {': ' + props.mobileNumber}
          </Text>
        </View>
      </Card.Content>
    </Card>
  );
};

export default DeliveryAddressCard;

export const DeliveryAddressCardCMSInput = {
  _component: CheckOutPageComponentType.DeliveryAddressCard,
  component: DeliveryAddressCard,
};
