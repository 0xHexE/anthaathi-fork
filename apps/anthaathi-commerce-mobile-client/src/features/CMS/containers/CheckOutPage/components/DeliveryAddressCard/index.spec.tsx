import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import DeliveryAddressCard from './index';

describe('DeliveryAddressCard', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddressCard
            deliveryTitle="Delivery Address"
            deliveryAddress="USA"
            buttonTitle="Change"
            mobileNumber="+90909090"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('deliveryAddressCard')).toBeTruthy();
  });

  it('should have DeliveryAddressCard title, address and mobile number', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddressCard
            deliveryTitle="Delivery Address"
            deliveryAddress="USA"
            buttonTitle="Change"
            mobileNumber="+90909090"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Delivery Address')).toBeTruthy();
    expect(temp.queryByText('USA')).toBeTruthy();
    expect(temp.queryByText('Mobile')).toBeTruthy();
    expect(temp.queryByText(': +90909090')).toBeTruthy();
  });

  it('should call when we call tap handlePress', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddressCard
            deliveryTitle="Delivery Address"
            deliveryAddress="USA"
            buttonTitle="Change"
            mobileNumber="+90909090"
            handlePress={onpress}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('changeAddressButton')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have DeliveryAddressCard button title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveryAddressCard
            deliveryTitle="Delivery Address"
            deliveryAddress="USA"
            buttonTitle="Change"
            mobileNumber="+90909090"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('deliverAddressButtonTitle')?.children).toContain(
      'Change',
    );
  });
});
