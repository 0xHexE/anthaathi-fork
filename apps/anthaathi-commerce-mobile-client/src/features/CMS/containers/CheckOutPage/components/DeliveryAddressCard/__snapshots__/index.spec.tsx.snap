// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryAddressCard should render the item 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "rgba(255, 251, 254, 1)",
          "borderColor": "#E3E2E7",
          "borderRadius": 4,
          "borderWidth": 1,
          "elevation": 1,
          "marginHorizontal": 10,
          "marginVertical": 5,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": 1,
            "flexShrink": 1,
          }
        }
        testID="deliveryAddressCard"
      >
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingVertical": 16,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "#364A15",
                  "fontSize": 16,
                  "fontWeight": "600",
                },
              ]
            }
          >
            Delivery Address
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#364A15",
                    "fontSize": 14,
                    "width": "85%",
                  },
                ]
              }
            >
              USA
            </Text>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "paddingHorizontal": 10,
                }
              }
              testID="changeAddressButton"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#008D3E",
                      "fontSize": 14,
                      "fontWeight": "500",
                    },
                  ]
                }
                testID="deliverAddressButtonTitle"
              >
                Change
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginVertical": 5,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#808080",
                    "fontSize": 14,
                  },
                ]
              }
            >
              Mobile
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "#364A15",
                    "fontSize": 14,
                  },
                ]
              }
            >
              : +90909090
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
